ec2/ec2.go

```diff
@@ -787,15 +777,35 @@ type IPPerm struct {
 	Protocol     string              `xml:"ipProtocol"`
 	FromPort     int                 `xml:"fromPort"`
 	ToPort       int                 `xml:"toPort"`
-	SourceIPs    []string            `xml:"ipRanges>item>cidrIp"`
+	SourceIPs    []SourceIP          `xml:"ipRanges>item"`
 	SourceGroups []UserSecurityGroup `xml:"groups>item"`
 }
 
+// SourceIP represents a single security group source IP address
+// within an ingress permission rule.
+//
+// See http://goo.gl/4oTxv for more details.
+type SourceIP struct {
+	IP string `xml:"cidrIp"`
+}
+
+// SourceIPs is a helper for constructing []SourceIP from strings.
+func SourceIPs(ips ...string) []SourceIP {
```

Comment:
This helper doesn't seem worth it. It's just a plain slice:

    SourceIP{{one}, {two}}